eos/b-decays/b-to-psd-psd-l-nu.cc

```diff
@@ -0,0 +1,503 @@
+        FormFactorFactory<PToPP>::option_specification(),
+        { "cp-conjugate"_ok, { "true", "false" },  "false" },
+        { "l"_ok,            { "e", "mu", "tau" }, "mu"    },
+        { "U"_ok,            { "c", "u" },         "c"     },
+        { "q"_ok,            { "u", "d", "s" },    "d"     },
+        { "I1"_ok,           { "1", "0", "1/2" },  "1"     },
+        { "I2"_ok,           { "1", "0", "1/2" },  "1"     },
+        { "C"_ok,            { "+-", "00", "+0" }, "+-"   },
+    };
+
+    const std::map<std::tuple<QuarkFlavor, QuarkFlavor, std::string, std::string, std::string>,
+                   std::tuple<std::string, std::string, std::string, std::string, std::string, IsospinRepresentation, IsospinRepresentation>>
+    Implementation<BToPPLeptonNeutrino>::Implementation::process_map
+    {
+        // Here we already squared them....
+        { { QuarkFlavor::up,    QuarkFlavor::up,    "1",    "1",    "00"   },
+          { "B->pipi",  "pipi->pipi", "B_u",  "pi^0", "pi^0", IsospinRepresentation::zero, IsospinRepresentation::one } },
+        { { QuarkFlavor::up,    QuarkFlavor::up,    "1",    "1",    "+-"   },
+          { "B->pipi",  "pipi->pipi", "B_u",  "pi^+", "pi^+", IsospinRepresentation::zero, IsospinRepresentation::one } },
+        { { QuarkFlavor::down,  QuarkFlavor::up,    "1",    "1",    "+0"   },
+          { "B->pipi",  "pipi->pipi", "B_d",  "pi^+", "pi^0", IsospinRepresentation::zero, IsospinRepresentation::one } },
+    };
+
+    BToPPLeptonNeutrino::BToPPLeptonNeutrino(const Parameters & p, const Options & o) :
+        PrivateImplementationPattern<BToPPLeptonNeutrino>(new Implementation<BToPPLeptonNeutrino>(p, o, *this))
+    {
+    }
+
+    BToPPLeptonNeutrino::~BToPPLeptonNeutrino()
+    {
+    }
+
+    double BToPPLeptonNeutrino::double_differential_decay_width(const double & q2, const double & k2) const
+    {
+        return _imp->differential_angular_observables(q2, k2).double_differential_decay_width();
+    }
+
+    double BToPPLeptonNeutrino::double_differential_mesonic_afb(const double & q2, const double & k2) const
+    {
+        return _imp->differential_angular_observables(q2, k2).double_differential_mesonic_afb();
+    }
+
+    double BToPPLeptonNeutrino::double_differential_branching_ratio(const double & q2, const double & k2) const
+    {
+        return _imp->differential_angular_observables(q2, k2).double_differential_decay_width() * _imp->tau_B / _imp->hbar;
+    }
+
+    double BToPPLeptonNeutrino::integrated_branching_ratio(const double & q2_min, const double & q2_max, const double & sqrt_k2_min, const double & sqrt_k2_max) const
+    {
+        std::function<double(const std::array<double, 2u> &)> integrand = [&] (const std::array<double, 2u> & x)
+        {
+            // Multiply by s -> sqrt(s) Jacobian
+            return 2.0 * x[1] * this->double_differential_branching_ratio(x[0], x[1] * x[1]);
+        };
+
+        auto config_cubature = cubature::Config().epsrel(5e-3);
```

Comment:
Please check if this is allocating memory dynamically. If so, please allocate once in the observable constructor instead.